src/plugins/intel_gpu/src/plugin/remote_context.cpp

```diff
@@ -370,26 +100,191 @@ AnyMap ExecutionContextImpl::getParams() const {
     return ret;
 }
 
-std::string ExecutionContextImpl::getDeviceName() const noexcept {
-    auto devName = m_plugin.lock()->GetName();
-
-    auto engine_type = cldnn::engine_types::ocl;
-    auto runtime_type = cldnn::runtime_types::ocl;
-    try {
-        // Use actual runtime and engine types
-        cldnn::device_query device_query(engine_type, runtime_type);
-        auto all_devices = device_query.get_available_devices();
-        auto current_device = m_engine->get_device();
-
-        for (auto& kv : all_devices) {
-            if (current_device->is_same(kv.second))
-                return devName + "." + kv.first;
+// For external contexts we try to match underlying handles with default contexts created by plugin to find device name
+std::string RemoteContextImpl::get_device_name(const std::vector<RemoteContextImpl::Ptr>& known_contexts,
+                                               const cldnn::device::ptr current_device) {
+    std::string device_name = "GPU";
```

Comment:
why do we have a hardcoded `GPU` here? What if plugin name is set differently by user or proxy ?